pyspinw/basis.py

```diff
@@ -0,0 +1,75 @@
+""" Functions for working with different basis vectors"""
+
+import numpy as np
+from pyspinw.dimensionality import dimensionality_check
+
+@dimensionality_check(vectors=(-1, 3))
+def find_aligned_basis(vectors: np.ndarray, rcond: float | None = None) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """ Find a set of orthonormal basis vectors aligned with the first being aligned to the input vectors
+
+    :param vectors: an n-by-3 numpy array containing the vectors to use to calculate the basis
+    :param rcond: tolerance on the calculation, default is 3 * machine precision
+    """
+
+    if rcond is None:
+        rcond = 3*np.finfo(vectors.dtype).eps # N * floating point epsilon
+
+    # Lengths, for checking for zeros and for normalising
+    lengths = np.sqrt(np.sum(vectors**2, axis=1))
```

Comment:
What about using `np.linalg.norm(vectors, axis=1)` instead? It's shorter and more understandable...

Reply:
done